nym-wallet/src/pages/bonding/node-settings/node-test/worker.ts

```diff
@@ -0,0 +1,101 @@
+// Copyright 2020-2023 Nym Technologies SA
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/* eslint-disable no-restricted-globals */
+import { NymNodeTester, set_panic_hook, current_network_topology, NodeTestResult } from 'nym-client-wasm';
+import { Network } from 'src/types';
+import { NodeTestEvent } from './types';
+
+console.log('Initializing worker');
+
+const postMessage = (data: NodeTestEvent) => self.postMessage(data);
+
+postMessage({
+  kind: 'WorkerLoaded',
+});
+
+const printAndDisplayTestResult = (result: NodeTestResult) => {
+  result.log_details();
+
+  postMessage({
+    kind: 'DisplayTesterResults',
+    args: {
+      result: {
+        score: result.score(),
+        sentPackets: result.sent_packets,
+        receivedPackets: result.received_packets,
+        receivedAcks: result.received_acks,
+        duplicatePackets: result.duplicate_packets,
+        duplicateAcks: result.duplicate_acks,
+      },
+    },
+  });
+};
+
+const buildTester = async (network: Network) => {
+  const validator = network === 'QA' ? 'https://qa-nym-api.qa.nymte.ch/api' : 'https://validator.nymtech.net/api/';
```

Comment:
we should use some global constants for these URLs, at least declared at the top of the file